sdk/eventhub/Azure.Messaging.EventHubs/tests/Amqp/AmqpErrorTests.cs

```diff
@@ -0,0 +1,471 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text.RegularExpressions;
+using Azure.Messaging.EventHubs.Errors;
+using Microsoft.Azure.Amqp;
+using Microsoft.Azure.Amqp.Encoding;
+using Microsoft.Azure.Amqp.Framing;
+using NUnit.Framework;
+
+namespace Azure.Messaging.EventHubs.Tests
+{
+    /// <summary>
+    ///   The suite of tests for the <see cref="AmqpError" />
+    ///   class.
+    /// </summary>
+    ///
+    [TestFixture]
+    public class AmqpErrorTests
+    {
+        /// <summary>
+        ///   The set of test cases for simple conditions and their matches
+        ///   to the proper exception types.
+        /// </summary>
+        ///
+        public static IEnumerable<object[]> SimpleConditionExceptionMatchTestCases()
```

Comment:
Cast all status codes to `int` here;  testing has shown that is how they are sent from the service.